modules/standard/Random.chpl

```diff
@@ -638,6 +638,30 @@ module Random {
         const oddseed = if seed % 2 == 0 then seed + 1 else seed;
         return oddseed;
       }
+      /*
+      Generate a seed based on the entropy pool in the OS; this requires
+      the presence of /dev/urandom.  This should be standard on both macOS
+      and Linux systems.
+
+      For highly parallel applications on one locale, it may be necessary
+      to lock this object to ensure two tasks do not both pull the same
+      bit of entropy.
+      */
+      proc type devURandomSeed: int(64) {
+        use IO;
+        var seed: int(64);
+        try {
+          var entropy = open('/dev/urandom', iomode.r);
```

Comment:
I recommend you use `openreader` for this. Either way, it's important to specify an `end` offset when creating the channel. Otherwise the channel will read-ahead (~4K) and the data will be thrown away (since you only use 8b).

Reply:
Re: the end offset, does the file object actually have that?  /dev/urandom is considered 0 bytes on macos, for instance.  Would Chapel be fine with that?

If not, that could certainly be another point in favor of using getrandom.

Reply:
The file doesn't have to have that many bytes. It just indicates when the channel needs to stop reading.

Reply:
(if the file has fewer bytes, you'd get EOF returned before that end offset)